cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -687,263 +816,737 @@ Allows setting index provider and index configuration for the backing index when
+CREATE INDEX [name] IF NOT EXISTS FOR ...
 ----
 a|
-New Cypher commands for listing constraints.
-
+Makes index creation idempotent. If an index with the name or schema already exists no error will be thrown.
 
 a|
-label:procedure[]
-label:deprecated[]
+label:syntax[]
+label:new[]
 [source, cypher, role="noheader"]
 ----
-db.indexes
+DROP INDEX name IF EXISTS
 ----
 a|
-Replaced by:
-[source, cypher, role="noheader"]
-----
-SHOW INDEXES
-----
-
+Makes index deletion idempotent. If no index with the name exists no error will be thrown.
 
 a|
-label:procedure[]
-label:deprecated[]
+label:syntax[]
+label:new[]
 [source, cypher, role="noheader"]
 ----
-db.indexDetails
+CREATE CONSTRAINT [name] IF NOT EXISTS ON ...
 ----
 a|
-Replaced by:
+Makes constraint creation idempotent. If a constraint with the name or type and schema already exists no error will be thrown.
+
+a|
+label:syntax[]
+label:new[]
 [source, cypher, role="noheader"]
 ----
-SHOW INDEXES YIELD *
+DROP CONSTRAINT name IF EXISTS
 ----
+a|
+Makes constraint deletion idempotent. If no constraint with the name exists no error will be thrown.
+
+|===
 
+[[cypher-deprecations-additions-removals-4.1]]
+== Version 4.1
+
+=== Restricted features
+
+[cols="2", options="header"]
+|===
+| Feature
+| Details
 
 a|
-label:procedure[]
-label:deprecated[]
+label:functionality[]
+label:restricted[]
 [source, cypher, role="noheader"]
 ----
-db.constraints
+REVOKE ...
 ----
 a|
-Replaced by:
+No longer revokes sub-privileges when revoking a compound privilege, e.g. when revoking `INDEX MANAGEMENT`, any `CREATE INDEX` and `DROP INDEX` privileges will no longer be revoked.
+
+a|
+label:functionality[]
+label:restricted[]
 [source, cypher, role="noheader"]
 ----
-SHOW CONSTRAINTS
+ALL DATABASE PRIVILEGES
 ----
+a|
+No longer includes the privileges `START DATABASE` and `STOP DATABASE`.
+|===
+
+=== Updated features
 
+[cols="2", options="header"]
+|===
+| Feature
+| Details
 
 a|
 label:procedure[]
-label:deprecated[]
+label:updated[]
 [source, cypher, role="noheader"]
 ----
-db.schemaStatements
+queryId
 ----
 a|
-Replaced by:
+The `queryId` procedure format has changed, and no longer includes the database name. For example, `mydb-query-123` is now `query-123`. This change affects built-in procedures `dbms.listQueries()`, `dbms.listActiveLocks(queryId)`, `dbms.killQueries(queryIds)` `and dbms.killQuery(queryId)`.
+
+a|
+label:functionality[]
+label:updated[]
 [source, cypher, role="noheader"]
 ----
-SHOW INDEXES YIELD *
+SHOW PRIVILEGES
 ----
+a|
+The returned privileges are a closer match to the original grants and denies, e.g. if granted `MATCH` the command will show that specific privilege and not the `TRAVERSE` and `READ` privileges. Added support for `YIELD` and `WHERE` clauses to allow filtering results.
+|===
+
+=== New features
+
+[cols="2", options="header"]
+|===
+| Feature
+| Details
+
+a|
+label:functionality[]
+label:new[] +
+For roles:
```

Comment:
we are introducing a new role so I'm not sure `For roles:` fits

Reply:
ok, I will just Role: if that makes more sense.

Reply:
that or `New role:` maybe?

Reply:
yes.